improver/cli/recursive_filter.py

```diff
@@ -57,9 +57,6 @@ def process(
     The smoothing_coefficient can be set on a grid square by grid-square basis
     for the x and y directions separately (using two arrays of
     smoothing_coefficients of the same dimensionality as the domain).
-    Smoothing coefficients can be generated using
-    :func:`~improver.utilities.ancillary_creation.OrographicSmoothingCoefficients`
-    and :func:`~improver.cli.generate_orographic_smoothing_coefficients`.
```

Comment:
I don't know enough about Sphinx configuration to answer this (see https://improver.readthedocs.io/en/latest/_modules/improver/cli.html#docutilize for code that does it), but what would you expect to be printed in the terminal in that case?

Reply:
Looking at the [clize source code](https://github.com/epsy/clize/blob/8e6312b/clize/help.py#L705-L717) there doesn't appear to be any way to configure the docutils parser it uses.

Reply:
I wouldn't expect an error to be raised when using the CLI help. I can get the optimal behaviour by adding the following line to the `docutilize` function, similar to #1041 i.e.
```
    doc = doc.replace(':func:', '')
```
This enables the hyperlinking within the CLI to work as intended without raising an error when running the CLI help.

The issue seems to be that `:func:` is not a valid restructured text role (https://docutils.sourceforge.io/docs/ref/rst/roles.html) but this role is valid when using the Sphinx [cross-referencing syntax](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects).

Reply:
This would print something like this:
```
Smoothing coefficients can be generated using ~improver.utilities.ancillary_creation.OrographicSmoothingCoefficients and
~improver.cli.generate_orographic_smoothing_coefficients.
```

While I would expect something more like this:
```
See also improver help generate-orographic-smoothing-coefficients.
```

We could add some regex kludge in `docutilize` to process these cross-references, but I'm not sure if this is worth the pain.

Reply:
Yes, I can see that trying to support both the CLI help and the Sphinx documentation is problematic. It seems like clize only currently supports the roles supported by `docutils`, rather than the roles used by sphinx. For reference, this [unanswered question](https://stackoverflow.com/questions/57765941/can-i-call-sphinx-parsers-parser-directly-and-parse-a-fragment-of-rest) seems related.  I assume that this would require some sort of change to clize or some over-riding of the clize default behaviour to work fully. For now, this change seems fine.